sentence.go

```diff
@@ -74,69 +78,125 @@ func (s BaseSentence) TalkerID() string {
 // String formats the sentence into a string
 func (s BaseSentence) String() string { return s.Raw }
 
-// parseSentence parses a raw message into it's fields
-func parseSentence(raw string) (BaseSentence, error) {
+var defaultSentenceParser = SentenceParser{
+	parserLock:    &sync.RWMutex{},
+	customParsers: map[string]ParserFunc{},
+}
+
+// SentenceParser provides configurable functionality to parse raw input into Sentence
+type SentenceParser struct {
+	parserLock    *sync.RWMutex
+	customParsers map[string]ParserFunc
```

Comment:
This is somewhat workaround with current implementation to have backwards compatibility.  This PR is backwards compatible so we still have these global public mehtods that can mutate customParsers.  In this PR we create global instance of "Special" `SentenceParser` instance that we use as global. And `Parse` is just

```go
func Parse(raw string) (Sentence, error) {
	return defaultSentenceParser.Parse(raw)
}
```

as `MustRegisterParser` and `RegisterParser` will mutate `defaultSentenceParser` istance `customParsers` map we need make sure that `Parse` method is co-routine safe because these `*RegisterParser` could potentially write while we are reading map.

Reply:
Right, the current functions should still be safe to access concurrently. But we can achieve that with a mutex outside the `SentenceParser` struct. For example:

``` go
var defaultSentenceParser SentenceParser
var defaultSentenceParserMu sync.Mutex

func Register(sentenceType string, parse ParseFunc) error {
    defaultSentenceParserMu.Lock()
    defer defaultSentenceParserMu.Unlock()
    return defaultSentenceParser.Register(sentenceType, parse)
}
```

Reply:
it is not that easy. `*Register` functions are only one side of concurrency.  only `Parse`  method implementation is moved to method on `SentenceParser` struct.   and `Parse` has that `parser, ok := p.config.CustomParsers[s.Type]` part that need lock.

Although we could move read lock to 
```go
func Parse(raw string) (Sentence, error) {
	defaultSentenceParserMu.Lock()
	defer defaultSentenceParserMu.Unlock()
	return defaultSentenceParser.Parse(raw)
}
```

but this would makes lock last "longer", as it locks more functionality, for all current users that are using global `Parse`

Reply:
Locking for "longer" doesn't affect performance if there's no contention.